include/cantera/kinetics/RxnRates.h

```diff
@@ -429,6 +533,141 @@ class ChebyshevRate
+
+    //! Add a coverage dependency for species *k*, with exponential dependence
+    //! *a*, power-law exponent *m*, and activation energy dependence *e*,
+    //! where *e* is in Kelvin, i.e. energy divided by the molar gas constant.
+    void addCoverageDependence(size_t k, doublereal a,
+                               doublereal m, doublereal e);
+
+    void update_C(const doublereal* theta) {
+        m_acov = 0.0;
+        m_ecov = 0.0;
+        m_mcov = 0.0;
+        size_t k;
+        doublereal th;
+        for (size_t n = 0; n < m_ac.size(); n++) {
+            k = m_sp[n];
+            m_acov += m_ac[n] * theta[k];
+            m_ecov += m_ec[n] * theta[k];
+        }
+        for (size_t n = 0; n < m_mc.size(); n++) {
+            k = m_msp[n];
+            th = std::max(theta[k], Tiny);
+            m_mcov += m_mc[n]*std::log(th);
+        }
+    }
+
+    /**
+     * Update the value of the rate constant.
+     *
+     * This function returns the actual value of the rate constant. It can be
+     * safely called for negative values of the pre-exponential factor.
+     */
+    doublereal updateRC(doublereal logT, doublereal recipT, doublereal deltaH) {
+        double Ea = activationEnergy_R(deltaH);
+        return m_A * std::exp(std::log(10.0)*m_acov + m_b*logT -
+                              (Ea)*recipT + m_mcov);
+    }
+
+    //! Return the pre-exponential factor *A* (in m, kmol, s to powers depending
+    //! on the reaction order) accounting coverage dependence.
+    /*!
+     *  Returns reaction pre-exponent accounting for both *a* and *m*.
+     */
+    doublereal preExponentialFactor() const {
+        return m_A * std::exp(std::log(10.0)*m_acov + m_mcov);
+    }
+
+    //! Return effective temperature exponent
+    doublereal temperatureExponent() const {
+        return m_b;
+    }
+
+    //! Return the activation energy divided by the gas constant (i.e. the
+    //! activation temperature) [K] based on the input enthalpy 
+    //! accounting coverage dependence.
+    doublereal activationEnergy_R(doublereal deltaH) {
+        double deltaH_R = deltaH / GasConstant; // deltaH in temperature units (Kelvin)
+        if (deltaH_R < -4 * m_E0) {
+            m_E = 0;
+        } else if (deltaH_R > 4 * m_E0) {
+            m_E = deltaH_R;
+        } else {
+            // m_w is in Kelvin
+            // vp is in Kelvin
+            double vp = 2 * m_w * ((m_w + m_E0) / (m_w - m_E0));
+            double vp_2w_dH = (vp - 2 * m_w + deltaH_R); // (Vp - 2 w + dH)
+            m_E = (m_w + deltaH_R / 2) * (vp_2w_dH * vp_2w_dH) / 
+                 (vp * vp - 4 * m_w * m_w + deltaH_R * deltaH_R); // in Kelvin
+        }
+        return m_E + m_ecov;
+    }
+
+    //! Return the intrinsic activation energy divided by the gas constant (i.e. the
+    //! activation temperature) [K], accounting coverage dependence.
+    doublereal activationEnergy_R0() const {
+        return m_E + m_ecov;
+    }
```

Comment:
Shouldn't this be `m_E0`, not `m_E`? In which case, similar logic applies and the `m_E` member variable could be eliminated for this class as well.